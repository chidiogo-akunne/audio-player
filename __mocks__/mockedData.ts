export const mockedData = {
  title: 'Nightlife',
  audio:
    'https://raw.githubusercontent.com/Learnfield-GmbH/CodingChallange/master/react%20native/simple%20audio%20player/data/Nightlife.mp3',
  cover:
    'https://raw.githubusercontent.com/Learnfield-GmbH/CodingChallange/master/react%20native/simple%20audio%20player/data/Nightlife.png',
  totalDurationMs: 15696
};
