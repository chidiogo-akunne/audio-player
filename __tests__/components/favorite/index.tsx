import Favorite from '@components/favorite';
import renderer from 'react-test-renderer';
import ThemeProvider from '@theme';
import { fireEvent, render } from '@testing-library/react-native';
import { ContextThemeProvider } from '@context/index';
import { mockedData } from '../../../__mocks__/mockedData';

describe('Favorite', () => {
  const tree = renderer.create(
    <ThemeProvider>
      <ContextThemeProvider>
        <Favorite item={mockedData} />
      </ContextThemeProvider>
    </ThemeProvider>
  );
  it(`renders correctly`, () => {
    expect(tree).toMatchSnapshot();
  });
  it(`it's clikable`, async () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <ThemeProvider>
        <ContextThemeProvider>
          <Favorite item={mockedData} />
        </ContextThemeProvider>
      </ThemeProvider>
    );
    await fireEvent.press(getByTestId('favorite'));
    expect(onPressMock.mock.calls.length);
  });
});
