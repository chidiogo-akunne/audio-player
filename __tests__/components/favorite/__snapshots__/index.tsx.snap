// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="favorite"
  >
    <Text />
  </View>
</View>
`;
